stl/inc/format

```diff
@@ -718,14 +719,34 @@ public:
     }
 };
 
+template <class _CharT>
+_NODISCARD constexpr _Fmt_codec<_CharT> _Get_fmt_codec() {
+    return {};
+}
+
+// clang-format off
+template <class _CharT>
+    requires (is_same_v<_CharT, char> && !_Is_execution_charset_self_synchronizing())
+_NODISCARD const _Fmt_codec<_CharT>& _Get_fmt_codec() {
+    // clang-format on
+    using _CodecType = _Fmt_codec<_CharT>;
+
+    static once_flag _Flag;
+    alignas(_CodecType) static unsigned char _Storage[sizeof(_CodecType)];
+
+    _STD call_once(_Flag, [] { _STD construct_at(reinterpret_cast<_CodecType*>(&_Storage)); });
+
+    return *_STD launder(reinterpret_cast<const _CodecType*>(&_Storage));

```

Comment:
I'm not sure whether laundering is necessary here (we aren't destroying and creating different objects), and there are no other occurrences in the entire STL. That said, I have no specific objection to this.

Reply:
Using `std::launder` here may be needed according to the Standard (per WG21-P0137), as an `unsigned char` array is not pointer-interconvertible with the object whose storage is provided by the array, and a plain `reinterpret_cast` can't obtain the pointer to the nested object.
However, AFAIK no compiler actually consider access to such a nested object without `std::launder` to be wrong.

Reply:
Ok, this is fine as-is then. Thanks for the explanation!